docs/code/MlNetCookBook.md

```diff
@@ -131,6 +131,39 @@ var reader = mlContext.Data.CreateTextReader(new[] {
 var data = reader.Read(dataPath);
 ```
 
+You can also create a data model class, and read the data based on this type. 
+
+```csharp
+// The data model. This type will be used through the document. 
+private class InspectedRow
+{
+	[LoadColumn(0)]
+	public bool IsOver50K { get; set; }
+
+	[LoadColumn(1)]
+	public string Workclass { get; set; }
+
+	[LoadColumn(2)]
+	public string Education { get; set; }
+
+	[LoadColumn(3)]
+	public string MaritalStatus { get; set; }
+
+	public string[] AllFeatures { get; set; }
```

Comment:
>AllFeatures [](start = 17, length = 11)

will this work? I am suspecting that reading this will throw

Reply:
it does work because i changed the logic. See my questions about whether we should process only the annotated members/field, to make the data models more usable. 

---
In reply to: [243104636](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243104636) [](ancestors = 243104636)